src/modules/ping_icmp.c

```diff
@@ -322,12 +338,31 @@ static int ping_icmp_handler(eventer_t e, int mask,
 
     /* If there is no timeout_event, the check must have completed.
      * We have nothing to do. */
-    if(!data->timeout_event) continue;
+    if(!data->timeout_event) {
+      mtevLT(nldeb, now,
+             "ping_icmp response timeout/completion for check '%s'\n", uuid_str);
+      continue;
+    }
 
     /* Sanity check the payload */
-    if(payload->check_no != data->check_no) continue;
-    if(payload->check_pack_cnt != data->expected_count) continue;
-    if(payload->check_pack_no >= data->expected_count) continue;
+    if(payload->check_no != data->check_no) {
+      mtevLT(nldeb, now,
+             "ping_icmp response check number mismatch for check '%s'\n", uuid_str);
+      continue;
+    }
+    if(payload->check_pack_cnt != data->expected_count) {
+      mtevLT(nldeb, now,
+             "ping_icmp response check pack count mismatch for check '%s'\n", uuid_str);
```

Comment:
Small nit: can we just say "packet" in the log message? I assume that is what the structure member refers to.  Same for line 360.

Reply:
sure